src/unix/os390-syscalls.c

```diff
@@ -103,17 +103,30 @@ static void maybe_resize(uv__os390_epoll* lst, unsigned int len) {
   unsigned int newsize;
   unsigned int i;
   struct pollfd* newlst;
+  struct epoll_event e;
 
   if (len <= lst->size)
     return;
 
+  /* Capture the message queue at the end */
+  if (lst->size == 0) {
+    e.fd = -1;
+  } else {
+    e.fd = lst->items[lst->size - 1].fd;
+    e.events = lst->items[lst->size - 1].events;
+    lst->items[lst->size - 1].fd = -1;
+  }
+
   newsize = next_power_of_two(len);
   newlst = uv__realloc(lst->items, newsize * sizeof(lst->items[0]));
 
   if (newlst == NULL)
     abort();
   for (i = lst->size; i < newsize; ++i)
     newlst[i].fd = -1;
+  /* Restore the message queue at the end */
+  newlst[newsize - 1].fd = e.fd;
+  newlst[newsize - 1].events = e.events;
```

Comment:
actually, they are 2 incompatible structs (pollfd, epoll_event) which is why this is needed. I will leave this as is